projects/plugins/jetpack/modules/comments/comments.php

```diff
@@ -479,104 +479,53 @@ public function watch_comment_parent() {
-					frame = document.getElementById( 'jetpack_remote_comment' ),
-					tellFrameNewParent;
-
-				tellFrameNewParent = function () {
-					if ( comm_par ) {
-						frame.src = "<?php echo esc_url_raw( $this->signed_url ); ?>" + '&replytocom=' + parseInt( comm_par, 10 ).toString();
-					} else {
-						frame.src = "<?php echo esc_url_raw( $this->signed_url ); ?>";
-					}
-				};
-
-				<?php if ( get_option( 'thread_comments' ) && get_option( 'thread_comments_depth' ) ) : ?>
-
-				if ( 'undefined' !== typeof addComment ) {
+			const iframe = document.querySelector( '#jetpack_remote_comment' );
+			<?php if ( get_option( 'thread_comments' ) && get_option( 'thread_comments_depth' ) ) : ?>
+			const watchReply = function() {
+				// Check addComment._Jetpack_moveForm to make sure we don't monkey-patch twice.
+				if ( 'undefined' !== typeof addComment && ! addComment._Jetpack_moveForm ) {
+					// Cache the Core function.
 					addComment._Jetpack_moveForm = addComment.moveForm;
-
-					addComment.moveForm = function ( commId, parentId, respondId, postId ) {
-						var returnValue = addComment._Jetpack_moveForm( commId, parentId, respondId, postId ),
-							cancelClick, cancel;
-
-						if ( false === returnValue ) {
-							cancel = document.getElementById( 'cancel-comment-reply-link' );
-							cancelClick = cancel.onclick;
-							cancel.onclick = function () {
-								var cancelReturn = cancelClick.call( this );
-								if ( false !== cancelReturn ) {
-									return cancelReturn;
-								}
-
-								if ( ! comm_par ) {
-									return cancelReturn;
-								}
-
-								comm_par = 0;
-
-								tellFrameNewParent();
-
-								return cancelReturn;
-							};
+					const commentParent = document.querySelector( '#comment_parent' );
+					const cancel = document.querySelector( '#cancel-comment-reply-link' );
+
+					function tellFrameNewParent ( commentParentValue ) {
+						const url = new URL( iframe.src );
+						if ( commentParentValue ) {
+							url.searchParams.set( 'replytocom', commentParentValue )
+						} else {
+							url.searchParams.delete( 'replytocom' );
 						}
-
-						if ( comm_par == parentId ) {
-							return returnValue;
+						if( iframe.src !== url.href ) {
+							iframe.src = url.href;
 						}
+					};
 
-						comm_par = parentId;
-
-						tellFrameNewParent();
+					cancel.addEventListener( 'click', function () {
+						tellFrameNewParent( false );
+					} );
 
-						return returnValue;
+					addComment.moveForm = function ( _, parentId ) {
+						tellFrameNewParent( parentId );
+						return addComment._Jetpack_moveForm.apply( null, arguments );
 					};
 				}
+			}
+			document.addEventListener( 'DOMContentLoaded', watchReply );
+			// In WP 6.4+, the script is loaded asynchronously, so we need to wait for it to load before we monkey-patch the functions it introduces.
+			document.querySelector('#comment-reply-js')?.addEventListener( 'load', watchReply );
```

Comment:
Just a note that listening for the `load` event is not 100% reliable, because an async script can (counter-intuitively) load in a blocking way if it is in the cache. Basically async means that the loading order is not defined. Nevertheless, since you're also monkey-patching at `DOMContentLoaded`, then this will be accounted for in the end.

Note also that this entire script could be made into an inline after script, using `wp_add_inline_script( 'comment-reply', $js_code, 'after' )`. This ensures it is printed with the `comment-reply` script rather than relying on the `wp_footer` action priority. Nevertheless, at the moment when this is done, core currently prevents `async` from applying on the script. We have [Core-58632](https://core.trac.wordpress.org/ticket/58632) open which would automatically delay such inline scripts of async/defer scripts. Nevertheless, for the time being the blocking behavior will apply.

Per above, better to use `getElementById` since faster than `querySelector`, when possible.

```suggestion
			document.getElementById('comment-reply-js')?.addEventListener( 'load', watchReply );
```

Reply:
Thanks Weston! I switched to `getElementById`